api/services/RStudioService.js

```diff
@@ -0,0 +1,117 @@
+var Promise = require('bluebird');
+var _ = require('lodash');
+cheerio = require('cheerio');
+
+module.exports = {
+	helpFindByTopicsAndPackages :function(topicNames,packageNames){
+		return Alias.orderedFindByTopicsAndPackages(topicNames,packageNames).then(function(results){
+			if (results.length == 0) return []; //no match found anywhere
+            if (results.length == 1) { //if there is only 1 match, redirect to this one (except if function is going to run on mutltiple aliases)
+        		return Topic.findOnePopulated({id: results[0].id}, {
+			        include: [{
+			          model: PackageVersion,
+			          as: 'package_version',
+			          attributes: ['package_name', 'version']
+			        }]
+			      }).then(function(topic) {
+			        if(topic === null) return [];
+			        else {
+			          return TopicService.computeLinks('/link/', topic)
+			            .then(function(topic) {
+			              topic.pageTitle = topic.name;
+			              return [topic];
+			            });
+			        }
+			    });
+			}
+            return results; //return all matches to list
+		});
+	},
+	helpFindByAlias:function(alias){
+		return Alias.orderedFindByAlias(alias).then(function(results){
```

Comment:
the `then` function seems to be very similar (or the same ?) than above.
Can you extract that behavior in another function that you would reuse in both places ?